logging/structured/src/main/scala/tofu/logging/impl/UniversalLogging.scala

```diff
@@ -0,0 +1,81 @@
+package tofu.logging
+package impl
+
+import cats.FlatMap
+import org.slf4j.{Logger, LoggerFactory, Marker}
+import tofu.logging.Logging.{Debug, Error, Info, Trace, Warn}
+import tofu.{Delay, WithContext}
+
+object UniversalLogging {
+  final def enabled(level: Logging.Level, logger: Logger): Boolean = level match {
+    case Trace => logger.isTraceEnabled()
+    case Debug => logger.isDebugEnabled()
+    case Info  => logger.isInfoEnabled()
+    case Warn  => logger.isWarnEnabled()
+    case Error => logger.isErrorEnabled()
+  }
+
+  final def write(level: Logging.Level, logger: Logger, message: String, values: Seq[LoggedValue]): Unit = level match {
+    case Trace => logger.trace(message, values: _*)
+    case Debug => logger.debug(message, values: _*)
+    case Info  => logger.info(message, values: _*)
+    case Warn  => logger.warn(message, values: _*)
+    case Error => logger.error(message, values: _*)
+  }
+  final def writeMarker(
+      level: Logging.Level,
+      logger: Logger,
+      marker: Marker,
+      message: String,
+      values: Seq[LoggedValue]
+  ): Unit                                                                                                =
+    level match {
+      case Trace => logger.trace(marker, message, values: _*)
+      case Debug => logger.debug(marker, message, values: _*)
+      case Info  => logger.info(marker, message, values: _*)
+      case Warn  => logger.warn(marker, message, values: _*)
```

Comment:
Those are impure as far as I see, so they should be marked as such somehow or even made private. I am like 99% sure that none of the tofu users will want to use the unsafe logger.